lib/UserAgentApplication.ts

```diff
@@ -123,14 +136,54 @@ namespace Msal {
                 this._cacheStorage.setItem(authorityKey, this.authority);
             }
 
-            const urlNavigate = authenticationRequest.createNavigateUrl(null);
+            const urlNavigate = authenticationRequest.createNavigateUrl(scopes) + "&prompt=select_account";
             this._loginInProgress = true;
-            if (this._interactionMode === this._interactionModes.popUp) {
-                this.openWindow(urlNavigate, "login", 20, this, this._tokenReceivedCallback, Constants.idToken);
-                return;
-            } else {
-                this.promptUser(urlNavigate);
-            }
+            this.promptUser(urlNavigate);
+
+        }
+
+        loginPopup(scopes: Array<string>, extraQueryParameters?: string): Promise<string> {
+            /*
+            1. Create navigate url
+            2. saves value in cache
+            3. redirect user to AAD
+            */
+            return new Promise<string>((resolve, reject) => {
+                if (this._loginInProgress) {
+                    reject("Login is in progress");
+                    return;
+                }
+
+                if (scopes) {
+                    const isValidScope = this.validateInputScope(scopes);
+                    if (isValidScope && !Utils.isEmpty(isValidScope)) {
+                        reject(isValidScope);
+                        return;
+                    }
+                }
+              
+                const authenticationRequest = new AuthenticationRequestParameters(this.authority, this.clientId, scopes, ResponseTypes.id_token, this.redirectUri);
+                if (extraQueryParameters) {
+                    authenticationRequest.extraQueryParameters = extraQueryParameters;
+                }
+
+                authenticationRequest.state = authenticationRequest.state + "|" + this.clientId;
+                this._cacheStorage.setItem(Constants.loginRequest, window.location.href);
+                this._cacheStorage.setItem(Constants.loginError, "");
+                this._cacheStorage.setItem(Constants.stateLogin, authenticationRequest.state);
+                this._cacheStorage.setItem(Constants.nonceIdToken, authenticationRequest.nonce);
+                this._cacheStorage.setItem(Constants.error, "");
+                this._cacheStorage.setItem(Constants.errorDescription, "");
+                const authorityKey = Constants.authority + Constants.resourceDelimeter + authenticationRequest.state;
+                if (Utils.isEmpty(this._cacheStorage.getItem(authorityKey))) {
+                    this._cacheStorage.setItem(authorityKey, this.authority);
+                }
+
+                const urlNavigate = authenticationRequest.createNavigateUrl(scopes) + "&prompt=select_account";
```

Comment:
>createNavigateUrl [](start = 58, length = 17)

Non blocking: do you agree that ideally this function should add the prompt=select_account qp?